src/Compiler.php

```diff
@@ -722,19 +751,31 @@ private function readObject()
         return true;
     }
 
-    private function readMap()
+    private function readFunction($which)
```

Comment:
This function allows behavior that isn't valid in some of the contexts where it is called.